src/systems/systems/src/air_conditioning/mod.rs

```diff
@@ -0,0 +1,387 @@
+        VariableIdentifier, Write,
+    },
+};
+
+use std::{collections::HashMap, time::Duration};
+
+use uom::si::{
+    f64::*, mass_rate::kilogram_per_second, pressure::hectopascal,
+    thermodynamic_temperature::degree_celsius,
+};
+
+pub mod acs_controller;
+pub mod cabin_air;
+
+pub trait DuctTemperature {
+    fn duct_demand_temperature(&self) -> HashMap<&'static str, ThermodynamicTemperature>;
+}
+
+pub trait FlowControlValveSignal {
+    fn should_open_fcv(&self) -> [bool; 2];
+}
+
+pub trait PackFlow {
+    fn pack_flow(&self) -> MassRate;
+}
+
+pub struct AirConditioningSystem {
+    acs_overhead: AirConditioningSystemOverhead,
+    acsc: ACSController,
+    pack_flow_valve: [PackFlowValve; 2],
+    // TODO: pack: [AirConditioningPack; 2],
+    // TODO: mixer_unit: MixerUnit,
+    // TODO: trim_air_system: TrimAirSystem,
+}
+
+impl AirConditioningSystem {
+    pub fn new(context: &mut InitContext, cabin_zone_ids: Vec<&'static str>) -> Self {
+        Self {
+            acs_overhead: AirConditioningSystemOverhead::new(context, &cabin_zone_ids),
+            acsc: ACSController::new(context, cabin_zone_ids),
+            pack_flow_valve: [
+                PackFlowValve::new(context, 1),
+                PackFlowValve::new(context, 2),
+            ],
+        }
+    }
+
+    pub fn update(
+        &mut self,
+        context: &UpdateContext,
+        engines: [&impl EngineCorrectedN1; 2],
+        pressurization: &impl CabinAltitude,
+        lgciu: [&impl LgciuWeightOnWheels; 2],
+    ) {
+        self.acsc.update(
+            context,
+            &self.acs_overhead,
+            &self.pack_flow_valve,
+            engines,
+            pressurization,
+            lgciu,
+        );
+
+        for fcv in self.pack_flow_valve.iter_mut() {
+            fcv.update(context, &self.acsc);
+        }
+    }
+}
+
+impl SimulationElement for AirConditioningSystem {
+    fn accept<V: SimulationElementVisitor>(&mut self, visitor: &mut V) {
+        self.acs_overhead.accept(visitor);
+        self.acsc.accept(visitor);
+        accept_iterable!(self.pack_flow_valve, visitor);
+
+        visitor.visit(self);
+    }
+}
+
+pub struct AirConditioningSystemOverhead {
+    pack_1_pb: OnOffFaultPushButton,
+    pack_2_pb: OnOffFaultPushButton,
+    temperature_selectors: HashMap<&'static str, ValueKnob>,
+}
+
+impl AirConditioningSystemOverhead {
+    fn new(context: &mut InitContext, cabin_zone_ids: &[&'static str]) -> Self {
+        let mut temperature_selectors: HashMap<&'static str, ValueKnob> = HashMap::new();
+        for id in cabin_zone_ids.iter() {
+            let knob_id = format!("COND_{}_SELECTOR", id);
+            temperature_selectors.insert(id, ValueKnob::new_with_value(context, &knob_id, 24.));
+        }
+        Self {
+            pack_1_pb: OnOffFaultPushButton::new_on(context, "COND_PACK_1"),
+            pack_2_pb: OnOffFaultPushButton::new_on(context, "COND_PACK_2"),
+            temperature_selectors,
+        }
+    }
```

Comment:
This is a good point, I didn't think about the speed of the `Hashmap`s. I've now refactored this and a couple of other instances where I was using `Hashmap` to use const generics, vectors and enums instead.

In this particular instance, I failed to use your suggestion because I couldn't find a way to implement Default for `ValueKnob` type (even if the default uses `None`, the type needs to implement `Copy`). I've used a vector instead initialised with a const generic array.